Source/Core/Core/LibDolphin.cpp

```diff
@@ -0,0 +1,458 @@
+					);
+
+				const u16 value = (const u16)(((u16)vbuffer[0] << 8) + ((u16)vbuffer[1]));
+				Memory::Write_U16(value, address);
+
+				char rbuffer[2];
+				rbuffer[0] = 0x16;
+
+				WriteFile(
+					h_pipe,
+					rbuffer,
+					2,
+					&bytes,
+					NULL
+					);
+				break;
+			}
+
+				// Write32
+			case 0x07:
+			{
+				char abuffer[4];
+				ReadFile(
+					h_pipe,
+					abuffer,
+					4,
+					&bytes,
+					NULL
+					);
+				const u32 address = ((u32)abuffer[0] << 24) + ((u32)abuffer[1] << 16) + ((u32)abuffer[2] << 8) + ((u32)abuffer[3]);
+
+				char vbuffer[4];
+				ReadFile(
+					h_pipe,
+					vbuffer,
+					4,
+					&bytes,
+					NULL
+					);
+
+				const u32 value = (const u32)(((u32)vbuffer[0] << 24) + ((u32)vbuffer[1] << 16) + ((u32)vbuffer[2] << 8) + ((u32)vbuffer[3]));
+				Memory::Write_U32(value, address);
+
+				char rbuffer[2];
+				rbuffer[0] = 0x17;
+
+				WriteFile(
+					h_pipe,
+					rbuffer,
+					2,
+					&bytes,
+					NULL
+					);
+				break;
+			}
+
+				// Write64
+			case 0x08:
+			{
+				char abuffer[4];
+				ReadFile(
+					h_pipe,
+					abuffer,
+					4,
+					&bytes,
+					NULL
+					);
+				const u32 address = ((u32)abuffer[0] << 24) + ((u32)abuffer[1] << 16) + ((u32)abuffer[2] << 8) + ((u32)abuffer[3]);
+
+				char vbuffer[8];
+				ReadFile(
+					h_pipe,
+					vbuffer,
+					8,
+					&bytes,
+					NULL
+					);
+
+				const u64 value = (const u64)(((u64)vbuffer[0] << 56) + ((u64)vbuffer[1] << 48) + ((u64)vbuffer[2] << 40) + ((u64)vbuffer[3] << 32) + ((u64)vbuffer[4] << 24) + ((u64)vbuffer[5] << 16) + ((u64)vbuffer[6] << 8) + ((u64)vbuffer[7]));
+				Memory::Write_U64(value, address);
+
+				char rbuffer[2];
+				rbuffer[0] = 0x18;
+
+				WriteFile(
+					h_pipe,
+					rbuffer,
+					2,
+					&bytes,
+					NULL
+					);
+				break;
+			}
+
+				// QueueInput
+			case 0x09:
+			{
+				char ibuffer[16];
+				ReadFile(
+					h_pipe,
+					ibuffer,
+					16,
+					&bytes,
+					NULL
+					);
+				u8 input[16];
+
+				input[0] = ibuffer[0];
+				input[1] = ibuffer[1];
+				input[2] = ibuffer[2];
+				input[3] = ibuffer[3];
+				input[4] = ibuffer[4];
+				input[5] = ibuffer[5];
+				input[6] = ibuffer[6];
+				input[7] = ibuffer[7];
+				input[8] = ibuffer[8];
+				input[9] = ibuffer[9];
+				input[10] = ibuffer[10];
+				input[11] = ibuffer[11];
+				input[12] = ibuffer[12];
+				input[13] = ibuffer[13];
+				input[14] = ibuffer[14];
+				input[15] = ibuffer[15];
+
+				struct GCPadStatus pad;
+				pad.button = ((u16)input[0] << 8) + ((u16)input[1]);
+				pad.stickX = input[2];
+				pad.stickY = input[3];
+				pad.substickX = input[4];
+				pad.substickY = input[5];
+				pad.triggerLeft = input[6];
+				pad.triggerRight = input[7];
+
+				pad.analogA = input[8];
+				pad.analogB = input[9];
+				pad.err = 0;
+
+				emu_pad->SetEmulatedInput(pad);
+
+				break;
+			}
+
+				// Frame advance
+			case 0x0A:
+			{
+				// I know it's bad practice, but we have to break from the while loop.
+				goto finish;
```

Comment:
I would consider a different design if this if this is what's required to end the loop.